test/math/FloatComplexTest.ooc

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright (c) 2011-2015 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use ooc-unit
+use ooc-math
+import math
+import text/StringTokenizer
+import structs/ArrayList
+import FloatComplex
+import lang/IO
+
+FloatComplexTest: class extends Fixture {
+
+	complexNumber0 := FloatComplex new (2,1)
+	complexNumber1 := FloatComplex new (3,2)
+	complexNumber2 := FloatComplex new (5,3)
+
+	init: func() {
+		super("FloatComplex")
+		this add("addition", func() {
+			expect((this complexNumber0 + this complexNumber1) real, is equal to(this complexNumber2 real))
+			expect((this complexNumber0 + this complexNumber1) imaginary, is equal to(this complexNumber2 imaginary))
+		})
+		this add("subtraction", func() {
+			expect((this complexNumber2 - this complexNumber1) real, is equal to(this complexNumber0 real))
+			expect((this complexNumber2 - this complexNumber1) imaginary, is equal to(this complexNumber0 imaginary))
+		})
+		this add("negative", func() {
+			expect((-this complexNumber0) real == -2, is true)
+			expect((-this complexNumber0) imaginary == -1, is true)
+		})
+		this add("mulitplication", func() {
+			expect((this complexNumber0 * this complexNumber1) real == 4, is true)
+			expect((this complexNumber0 * this complexNumber1) imaginary == 7, is true)
+		})
+		this add("scalar multiplication", func() {
+			expect(((-1.0f) * this complexNumber0) real, is equal to((-complexNumber0) real))
+			expect(((-1.0f) * this complexNumber0) imaginary, is equal to((-complexNumber0) imaginary))
+		})
+		this add("division", func() {
+			expect((complexNumber2 / complexNumber0) real == 2.6f, is true)
+			expect((complexNumber2 / complexNumber0) imaginary == 0.2f, is true)
+		})
+		// TODO: Add tests for compound assignment operators
+		this add("equality", func() {
+			expect(this complexNumber0 real == 2, is true)
+			expect(this complexNumber0 imaginary == 1, is true)
+			expect(this complexNumber0 == this complexNumber0, is true)
+			expect(this complexNumber0 != this complexNumber1, is true)
+			expect(this complexNumber1 == this complexNumber2, is false)
+		})
+		this add("toString", func() {
+			expect(this complexNumber0 toString(), is equal to("2.00 +1.00i"))
+			expect((FloatComplex parse("2.00 +1.00i")) == this complexNumber0, is true)
```

Comment:
Add tests for all combinations of positive and negative real and imaginary values, e.g. 
2+1i
2-1i
-2+1i
-2-1i
I'm pretty sure negative real values will break the `parse` method in its current implementation...